### ISSUE TITLE
Ruby 3.3.5 warnings

### ISSUE THREAD

**Issue by jaredcwhite** [bridgetownrb/bridgetown#924]
I got warnings that both `fiddle` and `rdoc` had to be added to the Gemfile because they won't be in future Ruby 3.5 stdlibs. Not sure yet what that's all about. Add to `bundle add` them both to remove the warnings.

**Comment by ianbayne** [comment]
I'm not sure about the `fiddle` warning, but the `rdoc` warning looks like it's coming from `irb` directly: https://bugs.ruby-lang.org/issues/20713#note-6. Checking that issue, I _think_ what's meant by [this](https://github.com/ruby/irb/pull/999#issuecomment-2331308387) comment is that if we were to add `irb` to the gemspec, we'd no longer get this warning.

I'd like to test this, but I'm not especially knowledgeable about gem development. I guess that I would need to update `bridgetown-core/bridgetown-core.gemspec` and re-build and re-install the gem. Doing so, however, I get symlink-related errors:

```bash
➜  gem install /bridgetown/bridgetown-builder-2.0.0.beta3.gem
ERROR:  While executing gem ... (Gem::Package::SymlinkError)
    installing symlink 'bridgetown-core/test/source/src/symlink-test/symlinked-file-outside-source' pointing to parent path /etc/passwd of /.rbenv/versions/3.3.5/lib/ruby/gems/3.3.0/gems/bridgetown-builder-2.0.0.beta3 is not allowed
```

I could reject all symlink files in the gemspec, something like:

```ruby
spec.files = `git ls-files -z`.split("\x0").reject { |f| File.symlink?(f) }
```

but I don't konw if that's the right way to go about this (spidey sense says no). Any help would be much appreciated 🙇

**Comment by ianbayne** [comment]
As an addition to the [above](https://github.com/bridgetownrb/bridgetown/issues/924#issuecomment-2480924485), note that I only saw the warnings about `fiddle` and `rdoc` when running `bin/bridgetown console`, which led to me suspecting that this is `irb`-related.

**Comment by jaredcwhite** [comment]
I'm getting different some similar warnings on Ruby 3.4.0-rc1. Looks like we'll need to add irb to the Bridgetown gemspec.

**Comment by jaredcwhite** [comment]
Resolved in 643f18229e495cd6fa532f4ace45f656ced52f33
